guilt-push: Avoid duplicate hits in a series

If a series file becomes mangled and a patch name appears in there
more than once, then the guilt-push will fail a horrible and
incomprehensible death.  Make it fail in a sensible way.

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

---

 guilt-push |    5 +++++
 1 file changed, 5 insertions(+)

--- a/guilt-push
+++ b/guilt-push
@@ -90,6 +90,11 @@ else
 	if [ -z "$eidx" ]; then
 		die "Patch $patch is not in the series or is guarded."
 	fi
+
+	matches=`echo $eidx | wc -w`
+	if [ $matches -gt 1 ]; then
+		die "Patch $patch is in the series multiple times"
+	fi
 fi
 
 # make sure that there are no unapplied changes
